mycroft/util/parse.py

```diff
@@ -164,6 +222,580 @@ def extractnumber_en(text):
+                        remainder = "am"
+                        used = 1
+                    elif int(word) > 100 and \
+                            (wordPrev == "o" or wordPrev == "oh"):
+                        # 0800 hours (pronounced oh-eight-hundred)
+                        strHH = int(word)/100
+                        strMM = int(word) - strHH*100
+                        military = True
+                        if wordNext == "hours":
+                            used += 1
+                    elif wordNext == "hours" and word[0] != '0' and \
+                            (int(word) < 100 and int(word) > 2400):
+                        # ignores military time
+                        # "in 3 hours"
+                        hrOffset = int(word)
+                        used = 2
+                        isTime = False
+                        hrAbs = -1
+                        minAbs = -1
+
+                    elif wordNext == "minutes":
+                        # "in 10 minutes"
+                        minOffset = int(word)
+                        used = 2
+                        isTime = False
+                        hrAbs = -1
+                        minAbs = -1
+                    elif wordNext == "seconds":
+                        # in 5 seconds
+                        secOffset = int(word)
+                        used = 2
+                        isTime = False
+                        hrAbs = -1
+                        minAbs = -1
+                    elif int(word) > 100:
+                        strHH = int(word)/100
+                        strMM = int(word) - strHH*100
+                        military = True
+                        if wordNext == "hours":
+                            used += 1
+                    elif wordNext[0].isdigit():
+                        strHH = word
+                        strMM = wordNext
+                        military = True
+                        used += 1
+                        if wordNextNext == "hours":
+                            used += 1
+                    elif wordNext == "" or wordNext == "o'clock" or \
+                            (wordNext == "in" and
+                                (wordNextNext == "the" or
+                                    wordNextNext == timeQualifier)):
+                        strHH = word
+                        strMM = 00
+                        if wordNext == "o'clock":
+                            used += 1
+                        if wordNext == "in" or wordNextNext == "in":
+                            used += (1 if wordNext == "in" else 2)
+                            if (
+                                    wordNextNext and
+                                    wordNextNext in timeQualifier or
+                                    (words[words.index(wordNextNext)+1] and
+                                    words[words.index(wordNextNext)+1] in timeQualifier)  # noqa
+                                ):
```

Comment:
In python there is no gain from having paranthesis around an entire if statement.